### PR TITLE
Support ingress annotations

### PR DESCRIPTION
**Pull request by schmichri**
# TL;DR
Based on [PR-188](https://github.com/lyft/flinkk8soperator/pull/188/files)

Supporting ingress annotations and tls Secrets. That allows the usage: 
* different ingress-controllers
* HTTPS with certmanager & letsencrypt
* Authentication on the ingress level

# Running Tests: 

## Apply CRD
```
kubectl apply -f https://raw.githubusercontent.com/iunera/flinkk8soperator/master/deploy/crd.yaml
```

## Dockerimage

Replace `image:` in `deploy/flinkk8soperator.yaml`
```
image: iunera/flinkk8soperator:0.5.1
```

## Ingress
Set some ingressAnnotations in the worcount example `examples/wordcount/flink-operator-custom-resource.yaml`

```
  ingressAnnotations:
    kubernetes.io/ingress.class: nginx
    kubernetes.io/tls-acme: "true"
    nginx.ingress.kubernetes.io/force-ssl-redirect: "true"

  ingressTLSSecretName: "wordcount-operator-example-tls"

```

### CODE REVIEW

**Comment by schmichri** [comment]
@anandswaminathan @mwylde @glaksh100 @premsantosh @maghamravi are you still shepherding this repo?

File: pkg/controller/flink/ingress.go [hunk 1]
```diff
@@ -58,6 +59,14 @@ func FetchJobManagerIngressCreateObj(app *flinkapp.FlinkApplication) *v1beta1.In
 			},
 		}},
 	}
+	if app.Spec.IngressTLSSecretName != "" {
+		ingressSpec.TLS = []v1beta1.IngressTLS{
+			{
+				Hosts:      []string{GetFlinkUIIngressURL(app.Name)},
```
Comment (maghamravi) [inline pkg/controller/flink/ingress.go:65]: nit:  `Hosts:      []string{GetFlinkUIIngressURL(getIngressName(app)},`   to ensure the app name honors the BG deployment mode.

**Comment by maghamravi** [comment]
Apologies for the delay in my review @schmichri !  Appreciate your effort in working on this feature !!

File: deploy/crd.yaml [hunk 2]
```diff
@@ -40,6 +40,12 @@ spec:
                 properties:
                   name:
                     type: string
+            ingressAnnotations:
+              description: May be used to configure the ingress objects
+              type: object
+            ingressTLSSecretName:
+              description: May be used to configure the ingress TLS
+              type: string
```
Comment (maccamlc) [inline deploy/crd.yaml:48]: Would an `ingressClassName` be useful here too? I've seen that the annotation class name is deprecated now.

**Comment by maccamlc** [comment]
I would like to use this feature, if it can be merged :)

**Pull request closed without merging**